src/app/Library/CrudPanel/Traits/Update.php

```diff
@@ -113,69 +115,69 @@ private function getModelAttributeValueFromRelationship($model, $field)
                 if (! isset($field['subfields'])) {
                     return $related_model->{$relation_method};
                 }
-                // we want to exclude the "self pivot field" since we already have it.
-                $pivot_fields = Arr::where($field['subfields'], function ($item) use ($field) {
-                    return $field['name'] != $item['name'];
-                });
 
                 $related_models = $related_model->{$relation_method};
                 $result = [];
 
-                // for any given model, we grab the attributes that belong to our pivot table.
                 foreach ($related_models as $related_model) {
                     $item = [];
                     switch ($relation_type) {
                         case 'HasMany':
                         case 'MorphMany':
-                            $result[] = $related_model->getAttributes();
+                            // we will get model direct attributes and merge with subfields values.
+                            $directAttributes = $related_model->withFakes()->getAttributes();
+                            $result[] = array_merge($directAttributes, $this->getSubfieldsValues($field['subfields'], $related_model));
                             break;
 
                         case 'BelongsToMany':
                         case 'MorphToMany':
+                            // for any given model, we grab the attributes that belong to our pivot table.
                             $item = $related_model->pivot->getAttributes();
-                            $item[$field['name']] = $related_model->getKey();
+                            $item[$relation_method] = $related_model->getKey();
                             $result[] = $item;
                             break;
                     }
                 }
 
                 return $result;
-
                 break;
             case 'HasOne':
             case 'MorphOne':
                 if (! method_exists($related_model, $relation_method)) {
                     return;
                 }
 
-                $related_entry = $related_model->{$relation_method};
+                $related_entry = $related_model->{$relation_method}->withFakes();
```

Comment:
Hmmm... what is the purpose of adding fakes here? I'm wondering... what if the related model does NOT have the `CrudTrait` on it? Wouldn't this stop it from working?